### PR TITLE
[11.0][IMP] stock_picking_report_delivery_note

### PR DESCRIPTION
**Pull request by mariadforgeflow**
Fix add lot quantities (when state done they weren't displayed)
@eantones

### CODE REVIEW

**Review by MiquelRForgeFlow — APPROVED** [review]
but fetch, rebase and squash 😉

File: stock_picking_report_delivery_note/models/stock_move.py [hunk 1]
```diff
@@ -52,9 +52,12 @@ def _get_amounts(self):
...
     def get_line_lots(self):
         if self.product_id.tracking in ('lot', 'serial'):
-            lots=[]
+            lots = []
             for l in self.move_line_ids:
-                lots.append("%s (%s %s)" % (l.lot_id.name,str(l.product_qty),l.product_uom_id.name))
+                if l.state == 'assigned':
+                    lots.append("%s (%s %s)" % (l.lot_id.name, str(l.product_qty), l.product_uom_id.name))
+                elif l.state == 'done':
+                    lots.append("%s (%s %s)" % (l.lot_id.name, str(l.qty_done), l.product_uom_id.name))
```
Comment (eantones) [inline stock_picking_report_delivery_note/models/stock_move.py]: Can you please find a way to avoid duplicate code in lines 58 and 60? Thanks! :-)

**Review by eantones — CHANGES_REQUESTED** [review]
@mariadforgeflow

File: stock_picking_report_delivery_note/models/stock_move.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply stock_picking_report_delivery_note/models/stock_move.py]:
```suggestion
                qty = False
                if l.state == 'assigned':
                    qty = l.product_qty
                elif l.state == 'done':
                    qty = l.qty_done
                if qty is not False:
                    lots.append("%s (%s %s)" % (l.lot_id.name, str(qty), l.product_uom_id.name))
```

Comment (eantones) [reply stock_picking_report_delivery_note/models/stock_move.py]: I was thinking of something more compact like

```suggestion
                qty = {'assigned': l.product_qty, 'done': l.qty_done}
                if l.state in qty:
                    lots.append("%s (%s %s)" % (l.lot_id.name, str(qty[l.state]), l.product_uom_id.name))
```
What do you think?

Comment (MiquelRForgeFlow) [reply stock_picking_report_delivery_note/models/stock_move.py]: nice

**Pull request merged**